### ISSUE TITLE
Make the DiagonalEK1 work for SecondOrderODEs

### ISSUE THREAD

**Issue by nathanaelbosch** [nathanaelbosch/ProbNumDiffEq.jl#302]
In principle this is quite straight-forward, but the fact that SecondOrderODEs by default always stack `[du; u]` is a bit annoying for us as that results in weird covariance structure. For example, if we have isometric Kronecker covariances of the form $I_d \otimes \Sigma$ in our state-space model, then `[du; u]` has covariances of the form $\tilde{Sigma} \cdot I_d$, i.e. the order of the Kronecker changed. For BlockDiagonals its a bit weirder: Instead of having a BlockDiagonal matrix, which has dense blocks on its diagonal, we have a BlockMatrix densely filled with Diagonals.

So basically, properly resolving this issue might mean implementing a way to efficiently switch between the two orderings. Or in the extreme, it migh even mean changing the ordering we have so far, which should then make this and related things easier.
